@here/harp-mapview/lib/MapView.ts

```diff
@@ -2917,15 +2922,25 @@ export class MapView extends THREE.EventDispatcher {
         if (currentFrameEvent !== undefined) {
             endTime = PerformanceTimer.now();
 
+            const frameRenderTime = endTime - frameStartTime;
+
             currentFrameEvent.setValue("render.setupTime", setupTime! - frameStartTime);
             currentFrameEvent.setValue("render.cullTime", cullTime! - setupTime!);
             currentFrameEvent.setValue("render.textPlacementTime", textPlacementTime! - cullTime!);
             currentFrameEvent.setValue("render.drawTime", drawTime! - textPlacementTime!);
             currentFrameEvent.setValue("render.textDrawTime", textDrawTime! - drawTime!);
             currentFrameEvent.setValue("render.cleanupTime", endTime - textDrawTime!);
-            currentFrameEvent.setValue("render.frameRenderTime", endTime - frameStartTime);
+            currentFrameEvent.setValue("render.frameRenderTime", frameRenderTime);
+
+            // Initialize the fullFrameTime with the frameRenderTime If we also create geometry in this frame
+            // this number will be increased in the TileGeometryLoader.
+            currentFrameEvent.setValue("render.fullFrameTime", frameRenderTime);
+
+            // Add THREE.js statistics
+            stats.addWebGLInfo(this.m_renderer.info);
 
-            PerformanceStatistics.instance.storeFrameInfo(this.m_renderer.info);
+            // Add memory statistics (FIXME: )
```

Comment:
What exactly needs to be fixed here? It is unclear to me.

Reply:
ups, forgot to actually write it down.